gguf-py/gguf/constants.py

```diff
@@ -49,6 +49,7 @@ class LLM:
         EXPERT_WEIGHTS_SCALE              = "{arch}.expert_weights_scale"
         POOLING_TYPE                      = "{arch}.pooling_type"
         LOGIT_SCALE                       = "{arch}.logit_scale"
+        DECODER_START_TOKEN_ID            = "{arch}.decoder_start_token_id"
```

Comment:
Is there a specific reason why the `decoder_start_token_id` isn't with the rest of the tokenizer config (like e.g. `tokenizer.ggml.bos_token_id`)?

In what way is it different from `tokenizer.ggml.bos_token_id`? When is it used?

Reply:
Yes, it's different. It's not related to the tokenizer at all, it's a model parameter. Decoder start token is not a separate specific token like BOS, EOS or PAD. It's used in encoder-decoder models like T5 as an initial starting token of the autoregressive decoding process. The model creators decided to use one of the existing tokens as the decoder start token (PAD in case of T5) and id of this token is stored in this parameter.